rpcclient/infrastructure.go

```diff
@@ -458,11 +455,20 @@ out:
 		// disconnected closed.
 		select {
 		case msg := <-c.sendChan:
-			err := c.wsConn.WriteMessage(websocket.TextMessage, msg)
+			conn, _, err := c.balancer.NextConn()
 			if err != nil {
-				c.Disconnect()
+				// Failed to get any usable connection, lets break out.
+				log.Tracef("RPC client balancer error: %s", err)
+				c.balancer.NeedsClientRestart = true
```

Comment:
This assumes that `c.balancer` is specifically the concrete type of `RoundRobinBalance`.  It appears the intent was to have a `Balancer` interface which the rest of the code would be agnostic to.

This logic doesn't seem quite right to achieve that goal.